client/js/compute.js

```diff
@@ -684,6 +684,79 @@ const compute_ops = [
     sample: 'var a = randRange ${x} ${y} ${z}',
     call: function(v, x, y, z) { return v = Math.round(Math.floor((Math.random() * (y - x) / (z || 1))) * (z || 1) + x) },
     hash: '4758cba8f404cfe3ef0a300cebf3938a'
+  },
+  {
+    name: 'colorMono',
+    desc: 'returns a color value of #000000 or #ffffff, whichever provides better contrast to the provided color',
+    sample: 'var a = colorMono ${x}',
+    call: function(v, x) {
+      // Code created by https://chat.openai.com/chat
+      let r = parseInt(x.slice(1, 3), 16);
+      let g = parseInt(x.slice(3, 5), 16);
+      let b = parseInt(x.slice(5, 7), 16);
+      let luminance = (0.2126 * r + 0.7152 * g + 0.0722 * b) / 255;
+      return v = (luminance <= 0.5) ? '#ffffff' : '#000000';
+    },
+    hash: '123'
+  },
+  {
+    name: 'colorHue',
+    desc: 'returns a hex color value in the same hue but different luminance of the provided color to provide contrast',
+    sample: 'var a = colorHue ${x} ${y} ${z}',
+    call: function(v, x, y, z) {
+      // Code created by https://chat.openai.com/chat
+      const color = x.slice(1);
+      const r = parseInt(color.substring(0, 2), 16);
+      const g = parseInt(color.substring(2, 4), 16);
+      const b = parseInt(color.substring(4, 6), 16);
+      const luminance = (0.2126 * r + 0.7152 * g + 0.0722 * b) / 255;
+      let outputColor;
+      let modifier;
+      if (y == 1) {
+        y = 0.6;
+      }
+      if (z == -1) {
+        modifier = (luminance > 0.5) ? y : -y;
+      } else {
+        modifier = (luminance > 0.5) ? -y : y;
+      }      
+      const colors = [r, g, b];
+      const adjustedColors = [];
+      for (let i = 0; i < colors.length; i++) {
+        const adjustedColor = Math.round(Math.max(0, Math.min(colors[i] + modifier * 255, 255)));
+        adjustedColors.push(adjustedColor);
+      }
+      const [adjustedR, adjustedG, adjustedB] = adjustedColors;
+      return v = 
+        "#" +
+        adjustedR.toString(16).padStart(2, '0') +
+        adjustedG.toString(16).padStart(2, '0') +
+        adjustedB.toString(16).padStart(2, '0');
+    },
+    hash: '123'
+  },
+  {
+    name: 'rgbToHex',
+    desc: 'converts an rgb color in format rgb(5,2,37) to hex',
+    sample: 'var a = rgbToHex ${x}',
+    call: function(v, x) {
+      function componentFromStr(numStr, percent) {
+        var num = Math.max(0, parseInt(numStr, 10));
+        return percent ?
+            Math.floor(255 * Math.min(100, num) / 100) : Math.min(255, num);
+      };
+      var rgbRegex = /^rgb\(\s*(-?\d+)(%?)\s*,\s*(-?\d+)(%?)\s*,\s*(-?\d+)(%?)\s*\)$/;
+      var result, r, g, b, hex = "";
+      if ( (result = rgbRegex.exec(x)) ) {
+          r = componentFromStr(result[1], result[2]);
+          g = componentFromStr(result[3], result[4]);
+          b = componentFromStr(result[5], result[6]);
+  
+          hex = "#" + (0x1000000 + (r << 16) + (g << 8) + b).toString(16).slice(1);
+      }
+      return v = hex;
```

Comment:
I think that if `colorStr` is `rgb(a,b,c)` or `rgba(a,b,c,d)` that this will do the job:
```
colors = colorStr.match(/\d+/g).map(function(a){ return parseInt(a,10); });
hex = '#' + colors.map(c => c.toString(16)).join('');
```
I haven't looked at the other functions, but I suspect that a similar improvement is possible.

Reply:
I tried that and it didn't work.  But maybe I had a typo or something else happened. I'll try again.

Reply:
Take a look at https://jsfiddle.net/rxl50/wLp9kb41/

Reply:
It doesn't handle the case where one of the values is a single digit.

Reply:
`hex = '#' + colors.map(c => c.toString(16).padStart(2,"0")).join('');` solves that problem.